### ISSUE TITLE
Zoom recordings problem and location can

### ISSUE THREAD

**Issue by sherpa003** [ogra1/zoom-snap#62]

**Comment by ogra1** [comment]
i dont have noted any issues with recording over here yet, if that happens again, can you check with ```snap changes``` if something perhaps got updated underneath you during the meeting ? 

i *can* actually reproduce the file location issue, it is a bit ominous tough that the file location input field is not editable at all in the dialog (not much i can do about that, seems to be an app bug), the open/change buttons seem to invoke the wrong xdg-portal dialog outside of the snap, this will require some time to research ...

**Comment by dholbach** [comment]
Confirmed. I'm seeing the same issue.

**Comment by dsoasti** [comment]
I'm having the same issue, I can't record any session on Zoom.
After click on the record button, I get the preference dialog and can't open or change the target file where records should be saved.
Any ideas how to fix it?

**Comment by rubencabrera** [comment]
I'm having the same issue too. Is there a workaround?

**Comment by dsoasti** [comment]
I managed to fix it like this:
Updated the system usign "actualizaciones de sistema", it took a while to update, then uninstalled zoom, updated again and re-installed Zoom. It worked for me. Hope it works for others as well.

**Comment by jeissonh** [comment]
I had this issue. Uninstalled zoom and the problem got dramatically bigger: Snap removed my local recordings with no confirmation. As a professor, I lost several recorded lessons, and I lost all my confidence in Snap.

**Comment by castrojo** [comment]
@jeissonh I thought my recordings were lost as well, they were just buried in a versioned directory `~/snap/zoom-client/143/blah` instead of a common place so I had no idea they were in there. Might be worth taking a look!

**Comment by cwatts-dorking** [comment]
I've had exactly this problem twice.  What worked both times was a) uninstall zoom-client b) restart computer c) install zoom-client.
Now I come to look at it I have lost my recordings.  The only one I now have is the one I used to check that it was working.  The trouble is that you haven't got time to do that once the meeting has started which you wanted to record.

**Comment by jeissonh** [comment]
Thanks for your responses!

In [another forum](https://forum.snapcraft.io/t/11336/4), I learned that recent versions of snap store a backup when one removes an application. The command `snap saved` list them, and they can be restored with `snap restore <backup_id>`.

I did not retry to fix the storing location problem. I just downloaded the Zoom installer from the official website, and it worked as expected.

**Comment by Fuseteam** [comment]
I can confirm that Snaps indeed saves files in `~/snap/<app-name>/<version>`

**Comment by mbdevpl** [comment]
I was having an issue which might be related to this (see #92).

TLDR:

When you try to save the meeting chat, snap tries to save it in the same old folder even after after the snap is updated, and fails because it cannot write to old folder (in my case the snap recordings config said `.../snap/zoom-client/148/...` while the snap was at version 160.

I fixed things by opening the recordings folder "Change" dialog and selecting some random file (for some ungodly reason that dialog is a file picker, not a folder picker) and then Zoom auto-magically fixed the path to `.../snap/zoom-client/160/...` and now I can save the chat again.

Maybe this will help with saving recordings as well?

I assume reinstalling the snap fixes the issue because it also resets the configuration :thinking: But I totally agree with **jeissonh** that reinstalling a snap without backing up everything first is a very bad idea -- I once lost all saved games for a game that was installed via snap, lost all faith in the platform since. Maybe didn't impact my work but still was painful.

**Comment by CodePint** [comment]
+1 on the saved recordings directory being incorrect.

You can change it in the zoom client setting by navigating to: Settings > Recording > Store my recording at

![Screenshot from 2022-06-10 12-02-12](https://user-images.githubusercontent.com/33781924/173051841-e0974fcd-a0c5-49a5-a16c-056ec223802c.png)
